mne_bids/tests/test_utils.py

```diff
@@ -135,3 +166,38 @@ def test_infer_eeg_placement_scheme():
     raw.rename_channels({'P3': 'foo'})
     placement_scheme = _infer_eeg_placement_scheme(raw)
     assert placement_scheme == 'n/a'
+
+
+def test_find_matching_sidecar():
+    """Test finding a sidecar file from a BIDS dir."""
+    # First write a BIDS dir
+    output_path = _TempDir()
+    data_path = testing.data_path()
+    raw_fname = op.join(data_path, 'MEG', 'sample',
+                        'sample_audvis_trunc_raw.fif')
+
+    event_id = {'Auditory/Left': 1, 'Auditory/Right': 2, 'Visual/Left': 3,
+                'Visual/Right': 4, 'Smiley': 5, 'Button': 32}
+    events_fname = op.join(data_path, 'MEG', 'sample',
+                           'sample_audvis_trunc_raw-eve.fif')
+
+    raw = mne.io.read_raw_fif(raw_fname)
+    write_raw_bids(raw, bids_basename, output_path, events_data=events_fname,
+                   event_id=event_id, overwrite=False)
```

Comment:
Can you avoid code duplication here?

Reply:
I think an option would be to put `test_find_matching_sidecar` from `test_utils.py` to `test_write.py`

... this is already being done with `read_raw_bids` --> it SHOULD be tested in `test_read.py` ... but to avoid code duplication, it is being tested in `test_write.py`

I don't like either of the issues:
- duplication
- testing functions in a test-module of a different module

perhaps a solution would be to rewrite the testing suite so that
1. tests modules are executed in a fixed order
1. test functions write to a dedicated location instead of a TEMP_DIR
1. at the end of the final test module, we clean up the dedicated location

Reply:
I opened an issue on this, because this implies a big refactoring of our tests.

https://github.com/mne-tools/mne-bids/issues/212